cmd/fluxd/main.go

```diff
@@ -274,6 +309,18 @@ func main() {
 		// There is only one way we currently interpret a repo of
 		// files as manifests, and that's as Kubernetes yamels.
 		k8sManifests = &kubernetes.Manifests{}
+
+		crdClient, err := crd.NewForConfig(restClientConfig)
+		if err == nil {
```

Comment:
I guess this is meant to only check for err once, but I think it makes the code harder to read (since the GO convention is to check for errors first)

Reply:
I really despise that convention :-/, but you are right that it's a convention and going against it is surprising to the reader.